src/components/Home/index.js

```diff
@@ -0,0 +1,297 @@
+              // making a deep copy of searchHistory
+              var search_history = JSON.parse(JSON.stringify(searchHistory));
+
+              if (search_history.length < 5) {
+                search_history.unshift(location);
+              } else {
+                search_history.pop();
+                search_history.unshift(location);
+              }
+              setSearchHistory(search_history);
+              search_history = JSON.stringify(search_history);
+              localStorage.setItem("localSearchHistory", search_history);
+            }
+            mapWeatherConditionToAPI[result.weather[0].main] === undefined
+              ? setWeather(Default)
+              : setWeather(mapWeatherConditionToAPI[result.weather[0].main]);
+          }
+        },
+        (error) => {
+          setIsLoaded(true);
+          setError(error);
+        }
+      );
+  }, [city.name, searchHistory]);
+
+  useEffect(() => {
+    if (coordinates) {
+      fetch(
+        "https://api.openweathermap.org/data/2.5/onecall?lat=" +
+          coordinates.lat +
+          "&lon=" +
+          coordinates.lon +
+          "&exclude=minutely,daily&units=metric&appid=" +
+          process.env.REACT_APP_APIKEY
+      )
+        .then((res) => res.json())
+        .then(
+          (result) => {
+            setOnecallResults(result.hourly);
+            setIsOnecallLoaded(true);
+          },
+          (error) => {
+            setError(error);
+            setIsOnecallLoaded(true);
+          }
+        );
+    }
+  }, [coordinates]);
+
+  const fetchLatLong = (name) => {
+    fetch(
+      `https://api.mapbox.com/geocoding/v5/mapbox.places/${name}.json?access_token=${accessToken}`
+    )
+      .then((res) => res.json())
+      .then(
+        (result) => {
+          setCityData({
+            name,
+            longitude: result.features[0].center[0],
+            latitude: result.features[0].center[1],
+          });
+        },
+        (error) => {
+          setIsLoaded(true);
+          setError(error);
+        }
+      );
+  };
+
+  if (error) {
+    return <div>Error: {error.message}</div>;
+  } else {
+    return (
+      <>
+        <img className="logo" src={logo} alt="MLH Prep Logo"></img>
+        <div>
+          <video
+            autoPlay
+            muted
+            loop
+            style={{
+              position: "fixed",
+              width: "100vw",
+              height: "100vh",
+              left: 0,
+              top: 0,
+              opacity: isLoaded ? 1 : 0,
+              transition: "opacity, 2s ease-in-out",
+              zIndex: "-1",
+              objectFit: "cover",
+            }}
+            src={weather[0]}
+          />
+          <h2>Enter a city below 👇</h2>
+          <div id="city-typeahead-container">
+            <PlacesTypeahead
+              apiKey={process.env.REACT_APP_API_NINJAS_API_KEY}
+              onChange={(selected) => selected && selected.length > 0}
```

Comment:
I believe we should call `fetchLatLong()` here as well because this event is fired when a suggestion is chosen. It is possible that the user may use mouse-click to select and not `Enter` key.
```suggestion
              onChange={(selected) => selected && selected.length > 0 && fetchLatLong(selected[0])}
```

Reply:
calling the api and changing the state would overload the api and would also cause map to move around to different coordinates on every change which won't be the ideal behavior i assume

Reply:
The `onChange` event of `react-bootstrap-typeahead` is fired when there is a change in selection, e.g., a suggestion in the dropdown is selected. On the other hand, there is `onInputChange` and it is fired as we type. Reference: https://github.com/ericgio/react-bootstrap-typeahead/blob/master/docs/API.md#typeahead.

Reply:
Triggering the change on onChange would increase the no of unnecessary re-renders and will cause performance issues and bugs.

Reply:
> Triggering the change on onChange would increase the no of unnecessary re-renders and will cause performance issues and bugs.

Can you elaborate?

Reply:
So, when we select an option from the dropdown, that causes a complete rerender of the typeahead component which makes the useEffect lose track of the variables and throws an error. Also, the weather is also currently being fetched only when we press enter so I think it would be an ideal behaviour if both fetching and marking on the map take place together.